cmake/get_info.cmake

```diff
@@ -0,0 +1,310 @@
+SET(detailed_log "${CMAKE_BINARY_DIR}/detailed.log")
+SET(minimal_log  "${CMAKE_BINARY_DIR}/minimal.log")
+FILE(REMOVE ${detailed_log} ${minimal_log})
+
+MACRO(_both)
+  # Write to both log files:
+  FILE(APPEND ${detailed_log} "${ARGN}")
+  FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_detailed)
+  # Only write to detailed.log:
+  FILE(APPEND ${detailed_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_minimal)
+  # Only write to minimal.log:
+  FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+FUNCTION(build_type_spec log_type var_name)
+    if(CMAKE_BUILD_TYPE MATCHES "Release")
+        set(var_string "${var_name}_RELEASE")
+        set(upd_string "\n#        ${var_name}_RELEASE                   ${${var_string}} ")
+        FILE(APPEND ${log_type} ${upd_string})
+    endif()
+    if(CMAKE_BUILD_TYPE MATCHES "Debug" )
+        set(var_string "${var_name}_DEBUG")
+        set(upd_string "#       ${var_name}_DEBUG                 ${${var_string}} ")
+        FILE(APPEND ${log_type} "${upd_string}")
+    endif()
+    if(CMAKE_BUILD_TYPE MATCHES "RelWithDebInfo")
+        set(var_string "${var_name}_RELWITHDEBINFO")
+        set(upd_string "#       ${var_name}_RELWITHDEBINFO                      ${${var_string}} ")
+        FILE(APPEND ${log_type} "${upd_string}")
+    endif()
+    if(CMAKE_BUILD_TYPE MATCHES "MinSizeRel")
+        set(var_string "${var_name}_MINSIZEREL")
+        set(upd_string "#       ${var_name}_MINSIZEREL                  ${${var_string}} ")
+        FILE(APPEND ${log_type} "${upd_string}")
+    endif()
+ENDFUNCTION()
+
+_both(
+"
+
+
+"
+)
+
+IF("${GINKGO_GIT_SHORTREV}" STREQUAL "")
+    _both("
+#########################################################################
```

Comment:
suggestion: can we move the "block formatting" into a separate function, so there is only a single place that needs to be changed? Referring to the `##############` and `#      ` prefixes here.
Also, I really like the consistency of the CMake output with its `-- ` formatting. What would you think about replacing this by `-----------------------` as separators and `-- ` as line prefixes?

Reply:
Yes, `--` does look more uniform. I will look into how I can extract that into a function. Should be simple enough.